### PR TITLE
Introduce useWalletAPIServer hook

### PR DESCRIPTION
**Pull request by JunichiSugiura**
### 📝 Description

This PR provides `useWalletAPIServer` hook to easily set up server with React component. This change allows us to shares more logic between desktop and mobile.

### ❓ Context

- **Impacted projects**: `ledger-live-desktop`, `ledger-live-mobile`, `ledger-live-common` 
- **Linked resource(s)**: [`LIVE-4852`](https://ledgerhq.atlassian.net/browse/LIVE-4852) 

### ✅ Checklist

- [ ] **Test coverage** 
- [x] **Atomic delivery** 
- [x] **No breaking changes** 

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

### CODE REVIEW

**Comment by JunichiSugiura** [comment]
I bumped `wallet-api-server` to the latest. Compile succeeded but I still need to test.

File: libs/ledger-live-common/src/wallet-api/react.ts [hunk 1]
```diff
@@ -218,27 +220,32 @@ function usePermission(manifest: AppManifest): Permission {
...
+export function useConfig({
+  tracking,
+  wallet,
+}: Pick<ServerConfig, "tracking" | "wallet">): ServerConfig {
+  const userId = useEnv("USER_ID");
 
-  return transportRef.current;
+  return useMemo(
+    () => ({
+      appId: "ledger-live",
```
Comment (JunichiSugiura) [inline libs/ledger-live-common/src/wallet-api/react.ts:242]: 🤔 Not sure what to put. If it's from manifest, it have to be set after instantiating `WalletAPIServer`.

File: libs/ledger-live-common/src/wallet-api/react.ts [hunk 2]
```diff
@@ -103,3 +147,521 @@ export function useGetAccountIds(
...
+    signFlowInfos: {
+      canEditFees: boolean;
+      hasFeesProvided: boolean;
+      liveTx: Partial<Transaction>;
+    };
+    options: Parameters<WalletHandlers["transaction.sign"]>[0]["options"];
+    onSuccess: (signedOperation: SignedOperation) => void;
+    onError: (error: Error) => void;
+  }) => void;
+  "transaction.broadcast": (
```
Comment (chabroA) [inline libs/ledger-live-common/src/wallet-api/react.ts:188]: Shouldn't it be `"transaction.signAndBroadcast"`?

`transaction.broadcast` does not exist anymore and has been replaced by `transaction.signAndBroadcast`
cf. [here](https://github.com/LedgerHQ/wallet-api/blob/main/packages/client/src/modules/Transaction.ts)

File: apps/ledger-live-mobile/src/components/WebPlatformPlayer/WebViewV2.tsx [hunk 3]
```diff
@@ -62,21 +36,234 @@ import { track } from "../../analytics/segment";
...
+          params: {
+            message,
+            accountId: account.id,
+            onConfirmationHandler: onSuccess,
+            onFailHandler: onError,
+          },
+          onClose: onCancel,
+        });
+      },
+      "transaction.sign": ({
```
Comment (chabroA) [inline apps/ledger-live-mobile/src/components/WebPlatformPlayer/WebViewV2.tsx:128]: Why does mobile only have `"transaction.sign"` and not `"transaction.signAndBrodcast"` as well anymore?

**Review by chabroA — COMMENTED** [review]
Interesting first step. Looks like there is some mismatch with `"transaction.signAndBroadcast"` and the old `"transaction.broadcast"`.

Also, how can this be tested?

File: libs/ledger-live-common/src/wallet-api/react.ts (hunk 2 above)
Comment (JunichiSugiura) [reply libs/ledger-live-common/src/wallet-api/react.ts:188]: In short, `keyof UIHook != keyof WalletHandlers`.

`transaction.signAndBroadcast` uses `transaction.sign` + `transaction.broadcast` UI hook underneath. The reason why it isn't called `signAndBroadcast` is that `sign` and `broadcast` require different UI to trigger.

File: apps/ledger-live-mobile/src/components/WebPlatformPlayer/WebViewV2.tsx (hunk 3 above)
Comment (JunichiSugiura) [reply apps/ledger-live-mobile/src/components/WebPlatformPlayer/WebViewV2.tsx:128]: `transaction.broadcast` is a UI that triggers when broadcast is done (optimistic UI feedback). AFAIK, currently it only exists in desktop. Broadcast logic is already handled within the custom hook in live-common. The purpose of this refactoring is to extract all common logic from UI and leave only platform specific UI implementation in desktop/mobile workspace.

**Review by JunichiSugiura — COMMENTED** [review]
@chabroA I hope my comments make sense to you.

File: apps/ledger-live-mobile/src/components/WebPlatformPlayer/WebViewV2.tsx (hunk 3 above)
Comment (JunichiSugiura) [reply apps/ledger-live-mobile/src/components/WebPlatformPlayer/WebViewV2.tsx:128]: transaction.broadcast is a UI that triggers when broadcast is done (optimistic UI feedback). AFAIK, currently it only exists in desktop. Broadcast logic is already handled within the custom hook in live-common. The purpose of this refactoring is to extract all common logic from UI and leave only platform specific UI implementation in desktop/mobile workspace.

File: libs/ledger-live-common/src/wallet-api/react.ts (hunk 2 above)
Comment (JunichiSugiura) [reply libs/ledger-live-common/src/wallet-api/react.ts:188]: In short, keyof UIHook != keyof WalletHandlers.

transaction.signAndBroadcast uses transaction.sign + transaction.broadcast UI hook underneath. The reason why it isn't called signAndBroadcast is that sign and broadcast require different UI to trigger.

File: libs/ledger-live-common/src/wallet-api/react.ts [hunk 4]
```diff
@@ -472,15 +478,48 @@ export function useWalletAPIServer({
           })
       )
     );
-  }, [manifest, tracking, accounts, server, uiMessageSign]);
+  }, [accounts, manifest, server, tracking, uiMessageSign]);
 
   useEffect(() => {
-    if (!uiTxSign || !uiTxBroadcast) return;
+    if (!uiTxSign) return;
```
Comment (JunichiSugiura) [inline libs/ledger-live-common/src/wallet-api/react.ts:484]: @Justkant 🙏

File: libs/ledger-live-common/src/wallet-api/react.ts [hunk 5]
```diff
@@ -472,15 +478,48 @@ export function useWalletAPIServer({
...
+                  reject(error);
+                },
+              })
+            )
+        );
+
+        return Buffer.from(signedOperation.signature);
+      }
+    );
+  }, [accounts, manifest, server, tracking, uiTxSign]);
```
Comment (JunichiSugiura) [inline libs/ledger-live-common/src/wallet-api/react.ts:515]: @Justkant 🙏

**Review by Justkant — APPROVED** [review]

**Pull request merged**